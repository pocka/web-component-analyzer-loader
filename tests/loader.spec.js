const compiler = require('./compiler')

test('CustomElement', async () => {
  const stats = await compiler('custom-element.js')
  const output = stats.toJson().modules[0].source

  expect(output).toMatch(/^module\.exports={[\s\S]+}$/)
  expect(output).toMatchSnapshot()
})

test('lit-element', async () => {
  const stats = await compiler('lit-element.js')
  const output = stats.toJson().modules[0].source

  expect(output).toMatch(/^module\.exports={[\s\S]+}$/)
  expect(output).toMatchSnapshot()
})
