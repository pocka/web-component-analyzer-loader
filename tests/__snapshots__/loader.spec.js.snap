// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomElement 1`] = `
"module.exports={
  \\"version\\": \\"experimental\\",
  \\"tags\\": [
    {
      \\"name\\": \\"my-element\\",
      \\"attributes\\": [
        {
          \\"name\\": \\"attr1\\"
        },
        {
          \\"name\\": \\"attr2\\"
        }
      ],
      \\"properties\\": [
        {
          \\"name\\": \\"myProp\\",
          \\"type\\": \\"string\\",
          \\"default\\": \\"\\\\\\"foo\\\\\\"\\"
        },
        {
          \\"name\\": \\"value\\"
        }
      ]
    }
  ]
}"
`;

exports[`lit-element 1`] = `
"module.exports={
  \\"version\\": \\"experimental\\",
  \\"tags\\": [
    {
      \\"name\\": \\"my-element\\",
      \\"attributes\\": [
        {
          \\"name\\": \\"prop4\\",
          \\"type\\": \\"string\\",
          \\"default\\": \\"\\\\\\"hello\\\\\\"\\"
        },
        {
          \\"name\\": \\"prop-5\\"
        },
        {
          \\"name\\": \\"prop1\\",
          \\"type\\": \\"string\\"
        },
        {
          \\"name\\": \\"prop-two\\",
          \\"type\\": \\"number\\"
        }
      ],
      \\"properties\\": [
        {
          \\"name\\": \\"myProp\\",
          \\"type\\": \\"string\\",
          \\"default\\": \\"\\\\\\"myProp\\\\\\"\\"
        },
        {
          \\"name\\": \\"prop4\\",
          \\"attribute\\": \\"prop4\\",
          \\"type\\": \\"string\\",
          \\"default\\": \\"\\\\\\"hello\\\\\\"\\"
        },
        {
          \\"name\\": \\"prop5\\",
          \\"attribute\\": \\"prop-5\\"
        },
        {
          \\"name\\": \\"prop1\\",
          \\"attribute\\": \\"prop1\\",
          \\"type\\": \\"string\\"
        },
        {
          \\"name\\": \\"prop2\\",
          \\"attribute\\": \\"prop-two\\",
          \\"type\\": \\"number\\"
        },
        {
          \\"name\\": \\"prop3\\",
          \\"type\\": \\"boolean\\"
        }
      ]
    }
  ]
}"
`;
